backend/cmm_helpers.ml

```diff
@@ -404,6 +404,15 @@ let rec sub_int c1 c2 dbg =
         add_const (sub_int c1 c2 dbg) n1 dbg
       | c1, c2 -> Cop (Csubi, [c1; c2], dbg))
 
+let add_int_addr c1 c2 dbg = Cop (Cadda, [c1; c2], dbg)
+
+let add_int_ptr ~ptr_out_of_heap c1 c2 dbg =
+  (* The [add_int_addr] case is only used for string access, and it seems
+     unlikely that the optimizations done for [add_int] will apply.
```

Comment:
Even if it's maybe not that frequent, I think that at least the optimization for when the index / offset added is `0` might be useful ?

Reply:
Ah yes, good point!